<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SliderControl</class>
 <widget class="QWidget" name="SliderControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>791</width>
     <height>51</height>
    </rect>
   </property>
   <widget class="QPushButton" name="bacl">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton { 
    background-color: #607D8B; 
    color: white; 
    border: none; 
    padding: 8px 16px; 
    border-radius: 4px; 
    font-weight: bold;
}
QPushButton:hover { 
    background-color: #455A64; 
}</string>
    </property>
    <property name="text">
     <string>←</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Slider Control</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>80</y>
     <width>761</width>
     <height>221</height>
    </rect>
   </property>
   <property name="title">
    <string>Joint angle control</string>
   </property>
   <widget class="QLabel" name="A_servo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Basement (servo A)</string>
    </property>
   </widget>
   <widget class="QLabel" name="A_range">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>range: 0° - 180°</string>
    </property>
   </widget>
   <widget class="QSlider" name="A_slider">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>50</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="value">
     <number>90</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
    <property name="tickInterval">
     <number>18</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="A_spinBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>50</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="value">
     <number>90</number>
    </property>
   </widget>
   <widget class="QLabel" name="B_servo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Axis B (servo B)</string>
    </property>
   </widget>
   <widget class="QLabel" name="B_range">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>100</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>range: 0° - 77°</string>
    </property>
   </widget>
   <widget class="QSlider" name="B_slider">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>100</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>77</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
    <property name="tickInterval">
     <number>8</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="B_spinBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>100</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="maximum">
     <number>77</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="C_servo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Axis C (servo C)</string>
    </property>
   </widget>
   <widget class="QLabel" name="C_range">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>150</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>range: 140° - 180°</string>
    </property>
   </widget>
   <widget class="QSlider" name="C_slider">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>150</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>140</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="value">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
    <property name="tickInterval">
     <number>4</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="C_spinBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>150</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="minimum">
     <number>140</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="value">
     <number>180</number>
    </property>
   </widget>
   <widget class="QLabel" name="G_servo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>190</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Gripper (servo G)</string>
    </property>
   </widget>
   <widget class="QLabel" name="G_range">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>190</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>range: 0° - 37°</string>
    </property>
   </widget>
   <widget class="QSlider" name="G_slider">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>190</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>37</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
    <property name="tickInterval">
     <number>4</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="G_spinBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>190</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>37</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="command">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>330</y>
     <width>611</width>
     <height>61</height>
    </rect>
   </property>
   <property name="title">
    <string>Control commands</string>
   </property>
   <widget class="QPushButton" name="sendall">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>131</width>
      <height>28</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton { 
    background-color: #2196F3; 
    color: white; 
    font-weight: bold; 
    padding: 8px 16px; 
    border-radius: 4px; 
    border: none;
}
QPushButton:hover { 
    background-color: #1976D2; 
}</string>
    </property>
    <property name="text">
     <string>Send all angles</string>
    </property>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>20</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton { 
    background-color: #FF9800; 
    color: white; 
    font-weight: bold; 
    padding: 8px 16px; 
    border-radius: 4px; 
    border: none;
}</string>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="statue">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>20</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Get Status</string>
    </property>
   </widget>
   <widget class="QPushButton" name="limits">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>20</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Display limits</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="communication">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>400</y>
     <width>611</width>
     <height>150</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>150</height>
    </size>
   </property>
   <property name="title">
    <string>Communication Log</string>
   </property>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>571</width>
      <height>91</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QWidget" name="command_2" native="true">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>110</y>
      <width>471</width>
      <height>41</height>
     </rect>
    </property>
    <widget class="QPushButton" name="clearlog">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>10</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear Log</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
